### ISSUE TITLE
[bug] Ctrl+Break does not work in conan

### ISSUE THREAD

**Issue by igadmg** [conan-io/conan#13246]
### Environment details

* Operating System+version: Windows 10
* Compiler+version:
* Conan version: 2.0
* Python version: 3.11

When conan is stuck in downloading phase or in many other places Ctlr+Break is simply ignored by conan app. And there is no way to stop it, only terminate.

**Comment by memsharded** [comment]
Hi @yatagarasu25 

These are the current signals handled by Conan:
```python
 def ctrl_c_handler(_, __):
        print('You pressed Ctrl+C!')
        sys.exit(USER_CTRL_C)

    def sigterm_handler(_, __):
        print('Received SIGTERM!')
        sys.exit(ERROR_SIGTERM)

    def ctrl_break_handler(_, __):
        print('You pressed Ctrl+Break!')
        sys.exit(USER_CTRL_BREAK)

    signal.signal(signal.SIGINT, ctrl_c_handler)
    signal.signal(signal.SIGTERM, sigterm_handler)

    if sys.platform == 'win32':
        signal.signal(signal.SIGBREAK, ctrl_break_handler)
```
It seems that the signal is correctly configured.
I have also tried a couple of experiments, and it seems to be correctly breaking, and the message is output, so not sure what could be happening.
Maybe it could be that the ``requests`` library might not be honoring it, or something like that. 
Can you please try to break in different points of your commands, and see what happens?

**Comment by igadmg** [comment]
Yes looks like break signals are not (or very lazily) triggered when conan is downloading something.
Also version 2.0 does not report download progress anymore

**Comment by memsharded** [comment]
> es looks like break signals are not (or very lazily) triggered when conan is downloading something.

Then it totally looks a limitation of the ``requests`` library, which would be difficult to change, we don't have control over that, we are basically calling ``requests.get(...)``, if the internal implementation decides to not honor ``Ctrl+Break`` for some reason, there is little we can do.

If the rest of Conan is correctly breaking, then, I'd suggest to close this issue, as it is not in the Conan side.

> Also version 2.0 does not report download progress anymore

yes, this is known, we removed the progress bars, they were a bit too problematic, we will try to introduce some progress indicator later, specifically we want to better address parallelism before the progress indicators.

**Comment by memsharded** [comment]
> yes, this is known, we removed the progress bars, they were a bit too problematic, we will try to introduce some progress indicator later, specifically we want to better address parallelism before the progress indicators.

Progress has been added for large files that take some time to upload/download, with % transfer updates every 10 seconds.

Closing the issue, thanks for the feedback!
